From 06c18bf7202e4de33782b07941da594f91422715 Mon Sep 17 00:00:00 2001
From: Christoph Reiter <reiter.christoph@gmail.com>
Date: Tue, 6 Jan 2026 13:02:49 +0100
Subject: [PATCH 015/N] build: Define MS_DLL_ID for sys.winver on MINGW

MS_DLL_ID is used for sys.winver and Windows registry keys
---
 configure.ac | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/configure.ac b/configure.ac
index 3828042..f99c50b 100644
--- a/configure.ac
+++ b/configure.ac
@@ -5333,6 +5333,26 @@ then
         [Defined when any dynamic module loading is enabled.])
 fi
 
+# MS_DLL_ID is used for sys.winver and Windows registry keys.
+case $host in
+*-*-mingw*)
+	case $host in
+	i686*)
+		CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"${VERSION}-32\"'"
+		;;
+	armv7*)
+		CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"${VERSION}-arm32\"'"
+		;;
+	aarch64*)
+		CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"${VERSION}-arm64\"'"
+		;;
+	*)
+		CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"$VERSION\"'"
+		;;
+	esac
+	;;
+esac
+
 # MACHDEP_OBJS can be set to platform-specific object files needed by Python
 
 AC_SUBST([MACHDEP_OBJS])
